src/components/diagnosticDrv/diagnosticIntro.directive.js

```diff
@@ -0,0 +1,61 @@
+'use strict';
+
+angular.module('znk.infra-web-app.diagnosticDrv').directive('diagnosticIntro', ['DiagnosticIntroSrv', '$translatePartialLoader', '$log',
+    function DiagnosticIntroDirective(DiagnosticIntroSrv, $translatePartialLoader, $log) {
+
+    var directive = {
+        restrict: 'E',
+        scope: {
+            showInstructions: '=?'
+        },
+        templateUrl: 'components/diagnosticDrv/diagnosticIntro.template.html',
+        link: function link(scope) {
+
+            $translatePartialLoader.addPart('diagnosticDrv');
+
+            scope.d = {};
+
+            DiagnosticIntroSrv.getActiveData().then(function(activeData) {
+                if (!activeData || !activeData.id) {
+                    $log.error('DiagnosticIntroDirective: activeData id must exist!');
+                }
+                scope.d.activeId = activeData.id;
+                return DiagnosticIntroSrv.getConfigMap();
+            }).then(function(mapData) {
```

Comment:
like to chain two different promises which the second depend on the first, change?